### PR TITLE
ruby testing

### PR DESCRIPTION
**Pull request by mahtsham**
Tic Tac Toe Testing
In this pull request, we are using rspec for testing the code.

### CODE REVIEW

File: spec/test_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      expect(board.update(1, 'O')).to eql('O')
+    end
+  end
+  describe 'winner?' do
+    it 'checks the winner' do
+      expect(board.winner?).to eql(false)
+    end
+  end
+
+  describe 'valid_move' do
```
Comment (mricanho) [inline spec/test_spec.rb:38]: You made a good test here, but try to create another test if a player makes an invalid move, in this way your tests will be more complete 🥇

**Review by mricanho — CHANGES_REQUESTED** [review]
# 🔴  Changes Required 🔴  

Hello @mahtsham, you did a great job with your rspecs🔥, just a few changes are required:

- **Add more tests that cover negative scenarios (I provided you a hint in a comment below)** 

You can contact me on slack if you have any questions just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding! 🖥️ 👨🏽‍💻

File: spec/test_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+describe Board do
+  let(:board) { Board.new }
+  describe 'update' do
+    it 'update the move' do
+      expect(board.update(2, 'X')).to eql('X')
+    end
+  end
+  describe 'update' do
+    it 'update the move' do
+      expect(board.update(1, 'O')).to eql('O')
```
Comment (Dipeshtwis) [inline spec/test_spec.rb:10]: You have used many `describe` blocks for the same method. It should be wrapped in one describe block.

File: spec/test_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    end
+  end
+  describe 'update' do
+    it 'update the move' do
+      expect(board.update(1, 'O')).to eql('O')
+      expect(board.update(1, false)).to eql(false)
+    end
```
Comment (Dipeshtwis) [inline spec/test_spec.rb:15]: You have used more than one expected behavior in the same `it` block. It should be different `it` block.

File: spec/test_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  end
+  describe 'full?' do
+    it 'checks if board is full' do
+      expect(board.full?).to eql(false)
+    end
+  end
+  describe 'switch' do
+    it 'checks if board is full' do
+      expect(board.switch).to eql(nil)
```
Comment (Dipeshtwis) [inline spec/test_spec.rb]: Same **mistake** here.
You have used another describe block for the same `switch` method. It should be wrapped in one describe block.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
## Hi @mahtsham, I'm Dipesh your reviewer for this PR 👩‍🚀 

It's good to see that you have covered all the method and tested for negative scenario as well but, there is some change needed Inorder to approve. I have highlighted some specific in further comments also.

## For Approval

- [ ] Specs for each method should be wrapped in one `describe` block.

- [ ] you should have one `it` block for each expected behaviour.

- [ ] For now, I can't able to see how to run the test (rspec) in your readme. Please mention it in the readme file as well.

I have highlighted some of the code in the comments below.
## Let's fix the required change and request another code review.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional question).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Github](https://github.com/Dipeshtwis)

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @mahtsham, you did all the changes required for your last code reviewer, nice work!🚀 

You can merge this branch now.🥇 

You can contact me on slack if you have any questions just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️
